File: MinerApplication/MinerApplication/GoHomeAndSleep.cpp
#include "stdafx.h"
#include "GoHomeAndSleep.h"
#include "MiningForGold.h"
GoHomeAndSleep::GoHomeAndSleep()
{
}

GoHomeAndSleep::~GoHomeAndSleep()
{
}

void GoHomeAndSleep::Execute(Miner* miner)
{
	cout << "I am at home and Sleeping" << endl;
	cout << "Sleep ZZZ" << endl;
	miner->m_Tiredness--;
}

File: Miner Application(Unity)/Assets/Scripts/Miner/GoHomeAndSleep.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
[System.Serializable]
public class GoHomeAndSleep : State<Miner>
{
    public GoHomeAndSleep()
    {

    }
    public override void Execute(Miner agent)
    {
        
        agent.TargetLocation = new Vector3(4, 1, 0);
  
        Debug.Log("I am at home and Sleeping");
        Debug.Log("Sleep ZZZ");
        agent.m_Tiredness--;
    }
}
